### PR TITLE
fix: xyz script fix and other bugs

### PR DESCRIPTION
**Pull request by alvindaiyan**
## Description

- fixed xyz script when selecting checkpoint names in one of them
- fixed automatic been selected for vae then expose model not found error
- improved our process to ensure the right original process_images_inner method can be load back always
- fixed previous merge conflicts for refreshing endpoint selection

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: aws_extension/sagemaker_ui.py [hunk 1]
```diff
@@ -844,7 +844,7 @@ def create_ui(is_img2img):
                                                      label="Inference Job: Time-Type-Status-Uuid",
                                                      elem_id="txt2img_inference_job_ids_dropdown"
                                                      )
-                txt2img_inference_job_ids_refresh_button = modules.ui.create_refresh_button(inference_job_dropdown, origin_update_txt2img_inference_job_ids, lambda: {"choices": txt2img_inference_job_ids}, "refresh_txt2img_inference_job_ids")
+                txt2img_inference_job_ids_refresh_button = modules.ui.create_refresh_button(inference_job_dropdown, origin_update_txt2img_inference_job_ids, lambda: {"choices": txt2img_inference_job_ids, "value": None}, "refresh_txt2img_inference_job_ids")
                 # print(f"gr.Row() txt2img_inference_job_ids is {txt2img_inference_job_ids}")
```
Comment (alvindaiyan) [inline aws_extension/sagemaker_ui.py:848]: someone revert my change, changed it back

File: scripts/main.py [hunk 2]
```diff
@@ -440,7 +460,28 @@ def _process_args_by_plugin(self, script_name, arg):
 
         return arg, models
 
-    def _controlnet_args(self, script_name, arg) -> Dict[str, List[str]]:
+    def _xyz_args(self, script_name, arg, current_index, args) -> Dict[str, List[str]]:
+        if script_name != 'x/y/z plot':
+            return {}
+
+        if type(arg) is not list:
```
Comment (yike5460) [inline scripts/main.py:467]: move this logic down to check if arg exist first

File: scripts/main.py [hunk 3]
```diff
@@ -273,18 +274,33 @@ def encode_no_json(obj):
...
         # fixme: not handle batches yet
         from modules import shared
-        if shared.opts.sd_vae and shared.opts.sd_vae != 'None':
-            if shared.opts.sd_vae == 'Automatic':
-                models['VAE'] = [models['Stable-diffusion'][0]]
-            else:
-                models['VAE'] = [shared.opts.sd_vae]
+        # as discussed, we not support automatic for simplicity because the default is Automatic
+        # if user need, has to select a vae model manually in the setting page
+        if shared.opts.sd_vae and shared.opts.sd_vae != 'None' and shared.opts.sd_vae != 'Automatic':
```
Comment (yike5460) [inline scripts/main.py:284]: "if shared.opts.sd_vae and shared.opts.sd_vae not in ['None', 'Automatic']: " will be more flexible. And what's the context for automatic for 'as discussed', this should not leave in the comments

File: scripts/main.py [hunk 4]
```diff
@@ -273,18 +274,33 @@ def encode_no_json(obj):
...
+            models['VAE'] = [shared.opts.sd_vae]
 
+        from modules.processing import get_fixed_seed
 
+        seed = get_fixed_seed(p.seed)
+        subseed = get_fixed_seed(p.subseed)
         p.setup_prompts()
+
+        if type(seed) == list:
```
Comment (yike5460) [inline scripts/main.py:293]: what else type could seed be? string/dict? make sure the else branch can handle all corner case.

File: scripts/main.py [hunk 5]
```diff
@@ -440,7 +460,28 @@ def _process_args_by_plugin(self, script_name, arg):
...
+
+        if type(arg) is not list:
+            return {}
+
+        if not arg:
+            return {}
+
+        # 10 represent the checkpoint_name option for both img2img and txt2img
+        # ref: xyz_grid.py#204
+        if current_index - 2 < 0 or args[current_index - 2] != 10:
```
Comment (yike5460) [inline scripts/main.py:475]: more context and with marco to explain the number '2' and '10'

**Review by yike5460 — COMMENTED** [review]
some comments need to be clarified

**Review by HappyXY — APPROVED** [review]

File: scripts/main.py (hunk 4 above)
Comment (alvindaiyan) [reply scripts/main.py:293]: this is from original logic, no need to check other types

File: scripts/main.py (hunk 5 above)
Comment (alvindaiyan) [reply scripts/main.py:475]: details can be found in the attached python file

File: scripts/main.py (hunk 3 above)
Comment (alvindaiyan) [reply scripts/main.py]: fixed

**Pull request merged**